config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/PolicyAndFolderLoader.java

```diff
@@ -29,12 +30,16 @@
     private final PolicyConverterRegistry policyConverterRegistry;
     private final FileUtils fileUtils;
     private final IdGenerator idGenerator;
+    private final EntityTypeRegistry entityTypeRegistry;
+    private final JsonTools jsonTools;
 
     @Inject
-    PolicyAndFolderLoader(PolicyConverterRegistry policyConverterRegistry, FileUtils fileUtils, IdGenerator idGenerator) {
+    PolicyAndFolderLoader(PolicyConverterRegistry policyConverterRegistry, FileUtils fileUtils, IdGenerator idGenerator, EntityTypeRegistry entityTypeRegistry, JsonTools jsonTools) {
         this.policyConverterRegistry = policyConverterRegistry;
         this.fileUtils = fileUtils;
         this.idGenerator = idGenerator;
+        this.entityTypeRegistry = entityTypeRegistry;
+        this.jsonTools = jsonTools;
```

Comment:
JsonTools seemed a singleton and easily gettable from JsonTools.INSTANCE. Why do you want to inject via constructor?
Are you plan to mock it while running unit tests?

Reply:
i think injection is preferable than singleton instance irrespective of mocking the instance for junits